File: backend/src/main/java/com/example/demo/bean/ProblemDomain.java
package com.example.demo.bean;

public class ProblemDomain extends Node{
	private int no;	//领域编号
	private String name;	//领域名称
	private String shortname;	//名称缩写
	private String type;	//领域类型
	private String property;	//物理特性
	private int x;	//位置信息
	private int y;	
	private int h;	
	private int w;
	private String state;
	public int getNo() {
		return no;
	}
	public void setNo(int no) {
		this.no = no;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getShortname() {
		return shortname;
	}
	public void setShortname(String shortname) {
		this.shortname = shortname;
	}
	public String getType() {
		return type;
	}
	public void setType(String type) {
		this.type = type;
	}
	public String getProperty() {
		return property;
	}
	public void setProperty(String property) {
		this.property = property;
	}
	public int getX() {
		return x;
	}
	public void setX(int x) {
		this.x = x;
	}
	public int getY() {
		return y;
	}
	public void setY(int y) {
		this.y = y;
	}
	public int getH() {
		return h;
	}
	public void setH(int h) {
		this.h = h;
	}
	public int getW() {
		return w;
	}
	public void setW(int w) {
		this.w = w;
	}
	public String getState() {
		return state;
	}
	public void setState(String state) {
		this.state = state;
	}
}

File: frontend/src/app/entity/ProblemDomain.ts
import { Phenomenon } from './Phenomenon';
import { Node } from './Node'
export class ProblemDomain extends Node{
	no: number;	
	name: string;	
	shortname: string;	
	type: string;	
	property: string;	
	x: number;	
	y: number;	
	h: number;	
	w: number;
	phes: Phenomenon[]
	static newProblemDomain(no,name, shortname, type, property,x,y,w,h){
		let pd = new ProblemDomain()
		pd.no = no
		pd.name=name
		pd.shortname=shortname
		pd.type=type
		pd.property=property
		pd.x = x
		pd.y = y
		pd.h = h
		pd.w = w
		return pd
	}
	static newProblemDomainWithOld(old:ProblemDomain, name, shortname, type, property){
		let pd = new ProblemDomain()
		pd.no = old.no
		pd.x = old.x
		pd.y = old.y
		pd.h = old.h
		pd.w = old.w

		pd.name=name
		pd.shortname=shortname
		pd.type=type
		pd.property=property
		return pd
	}
	static copyProblemDomain(old:ProblemDomain){
		let pd = new ProblemDomain()
		pd.no = old.no
		pd.x = old.x
		pd.y = old.y
		pd.h = old.h
		pd.w = old.w

		pd.name=old.name
		pd.shortname=old.shortname
		pd.type=old.type
		pd.property=old.property
		return pd
	}
	getNo(){return this.no	}
    getName(){return this.name}
    getShortname(){return this.shortname}
	getX(){return this.x}
    getY(){return this.y}
    getH(){return this.h}
	getW(){return this.w}
	getProperty(){return this.property}
	getType(){return this.type}
	setNo(no){this.no=no}
    setName(name){this.name=name}
    setShortname(shortname){this.shortname=shortname}
	setX(x){this.x=x}
    setY(y){this.y=y}
    setH(h){this.h=h}
	setW(w){this.w=w}
	setProperty(property){this.property=property}
	setType(type){this.type=type}
}
